modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala

```diff
@@ -95,11 +95,13 @@ class Engine[D, U](stateL: Engine.State[D]) {
     x.map(p =>
       modifyS(c.sid)(_.startSingle(c.actCoords)) *>
       Handle.fromStream[D, EventType](
-        p.map{
-          case r@Result.OK(_)    => singleRunCompleted(c, r)
-          case e@Result.Error(_) => singleRunFailed(c, e)
-          case r                 =>
+        p.attempt.collect {
```

Comment:
You're using `collect` here to discard `Left(_: Throwable)`?

Reply:
Yes, do you think we should catch `Throwable` too?

Reply:
Discarding `Throwable`s makes me uneasy.

Reply:
I think we might want `case Left(NonFatal(e)) => ...` which will catch Throwables other than ones that mean you really do want the application to crash.

Reply:
Yeah that sounds reasonable. Using `collect` makes me a bit uneasy but `map` would need to do something with e.g. `OOM`